lightly/models/modules/center.py

```diff
@@ -74,20 +60,33 @@ def update(self, x: Tensor) -> None:
                 Feature tensor used to update the center. Must have the same number of
                 dimensions as self.size.
         """
-        batch_center = self._center_fn(x)
-        # Use copy for backwards compatibility with DINOLoss.
-        self.center.copy_(self._center_momentum(batch_center))
+        batch_center = self._center_fn(x=x, dim=self.dim)
+        self.center = center_momentum(
+            center=self.center, batch_center=batch_center, momentum=self.momentum
+        )
 
     @torch.no_grad()
     def _center_mean(self, x: Tensor) -> Tensor:
         """Returns the center of the input tensor by calculating the mean."""
-        batch_center = torch.mean(x, dim=self.dim, keepdim=True)
-        if dist.is_available() and dist.is_initialized():
-            dist.all_reduce(batch_center)
-            batch_center = batch_center / dist.get_world_size()
-        return batch_center
+        return center_mean(x=x, dim=self.dim)
 
-    @torch.no_grad()
-    def _center_momentum(self, batch_center: Tensor) -> Tensor:
-        """Returns the new center with momentum update."""
-        return self.center * self.momentum + batch_center * (1 - self.momentum)
+
+@torch.no_grad()
+def center_mean(x: Tensor, dim: Tuple[int, ...]) -> Tensor:
+    """Returns the center of the input tensor by calculating the mean."""
+    batch_center = torch.mean(x, dim=dim, keepdim=True)
+    if dist.is_available() and dist.is_initialized():
+        dist.all_reduce(batch_center)
+        batch_center = batch_center / dist.get_world_size()
+    return batch_center
+
+
+@torch.no_grad()
+def center_momentum(center: Tensor, batch_center: Tensor, momentum: float) -> Tensor:
+    """Returns the new center with momentum update."""
+    return center * momentum + batch_center * (1 - momentum)
+
+
+CENTER_MODE_TO_FUNCTION = {
```

Comment:
Why is this necessary? Do we plan to extend this in the future?

Reply:
Yes, DINOv2 to uses sinkhorn centering for some models